### PR TITLE
0.4.0

### PR DESCRIPTION
**Pull request by suzhoum**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -104,37 +104,37 @@ def default_setup_args(*, version):
...
 version = update_version(version, use_file_if_exists=False, create_file=True)
 
 install_requires = [
     # common module provides utils with stable api across minor version
-    "autogluon.common>=0.7,<1.1",
+    "autogluon.common>=0.7",
     # <2 because unlikely to introduce breaking changes in minor releases. >=1.10 because 1.10 is 3 years old, no need to support older
     "boto3>=1.10,<2.0",
-    "numpy>=1.21,<1.27",
+    "numpy>=1.21,<1.29",
```
Comment (tonyhoo) [inline setup.py:115]: We don't need this as core comes with numpy dependency: https://github.com/autogluon/autogluon/blob/master/core/src/autogluon/core/_setup_utils.py#L20

File: setup.py [hunk 2]
```diff
@@ -104,37 +104,37 @@ def default_setup_args(*, version):
...
     "packaging>=23.0,<24.0",
-    "pandas>=2.0.0,<2.2.0",
+    "pandas>=2.0.0,<2.3.0",
```
Comment (tonyhoo) [inline setup.py:117]: Same here

File: setup.py [hunk 3]
```diff
@@ -104,37 +104,37 @@ def default_setup_args(*, version):
...
     # updated sagemaker is required to fetch latest container info, so we don't want to cap the version too strict
     # otherwise cloud module needs to be released to support new container
     "sagemaker>=2.126.0,<3.0",
     "pyarrow>=11.0,<11.1",
     "PyYAML~=6.0",
-    "Pillow>=9.3.0,<10.0",  # unlikely to introduce breaking changes in minor releases
-    "ray[default]>=2.6.3,<2.7",
+    "Pillow>=10.2,<11",  # unlikely to introduce breaking changes in minor releases
```
Comment (tonyhoo) [inline setup.py:123]: same here

File: setup.py (hunk 1 above)
Comment (suzhoum) [reply setup.py:115]: Good point. Will update!

File: setup.py (hunk 3 above)
Comment (suzhoum) [reply setup.py:121]: I kept this as these are not dependencies in `autogluon.common`

**Review by tonyhoo — APPROVED** [review]

**Pull request merged**